pyrad/packet.py

```diff
@@ -563,15 +563,27 @@ def SaltCrypt(self, value):
             # self.authenticator = self.CreateAuthenticator()
             self.authenticator = 16 * six.b('\x00')
 
-        salt = struct.pack('!H', random_generator.randrange(0, 65535))
-        salt = chr(ord(salt[0]) | 1 << 7)+salt[1]
+        if six.PY3:
+            random_value = 32768 + random_generator.randrange(0, 32767)
```

Comment:
I don't think this is correct, this is effectively a random number between 32768 and 65535 when it should be between 0 and 65535 (the full range of 2 bytes). The random value calculation is the same for the python2 and python3 case, so move it please out of the conditional blocks.